src/lewis/adapters/stream.py

```diff
@@ -37,27 +37,60 @@ class StreamHandler(asynchat.async_chat):
     def __init__(self, sock, target, stream_server):
         asynchat.async_chat.__init__(self, sock=sock)
         self.set_terminator(b(target.in_terminator))
-        self.target = target
-        self.buffer = []
+        self._readtimeout = target.readtimeout
+        self._readtimer = 0
+        self._target = target
+        self._buffer = []
 
         self._stream_server = stream_server
 
         self._set_logging_context(target)
         self.log.info('Client connected from %s:%s', *sock.getpeername())
 
+    def process(self, msec):
+        if len(self._buffer) > 0:
```

Comment:
You're right, if buffer is empty the next condition should not be True either, so nothing should happen. Making it return early would make that much clearer. :+1: